File: week6-Python/Problem6/less/sentimental-cash/cash.py
import cs50




def main():

    # Here it prompts the user for the cash owned
    cents = get_cents()

    # Calculates the number of quarters in the cash owned
    quarters = calculate_quarters(cents)
    # Then it subsctract all quarters from the cash owned then now "cents" is the cash owned left without containing any quarter
    cents = cents - quarters * 25

    # Calculates the number of dimes in the cash owned
    dimes = calculate_dimes(cents)
    # Then it subsctract all dimes from the cash owned then now "cents" is the cash owned left without containing any dimes
    cents = cents - dimes * 10

    # Calculates the number of nickels in the cash owned
    nickels = calculate_nickels(cents)
    # Then it subsctract all nickels from the cash owned then now "cents" is the cash owned left without containing any nickels
    cents = cents - nickels * 5

    # Calculates the number of pennies in the cash owned
    pennies = calculate_pennies(cents)
    # Then it subsctract all pennies from the cash owned then now "cents" is the cash owned left without containing any pennies
    cents = cents - pennies * 1

    # Finally it adds all the coins types to each other, then prints the coins cowned !
    coins = quarters + dimes + nickels + pennies
    print(f'{int(coins)}')






def get_cents():

    # This will keep prompting untill the user gives valid answer
    while True:
        dollars = cs50.get_float('change owned: ')
        if dollars >= 0:
            break
    # Converts the change/cash owned typed by user from dollars to cents
    cents = round(dollars * 100)
    return cents

def calculate_quarters(cents):
    remainder = cents % 25
    quarters = (cents - remainder) / 25
    return quarters

def calculate_dimes(cents):
    remainder = cents % 10
    dimes = (cents - remainder) / 10
    return dimes

def calculate_nickels(cents):
    remainder = cents % 5
    nickels = (cents - remainder) / 5
    return nickels

def calculate_pennies(cents):
    remainder = cents % 1
    pennies = (cents - remainder) / 1
    return pennies









main()

File: week1-C/Problem1/less/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    int ChangeOwned;
    do
    {
        ChangeOwned = get_int("Change Owned: ");
    }
    while (ChangeOwned < 0);

    return ChangeOwned;
}

int calculate_quarters(int cents)
{
    int quarters = 0, i;

    for (i = 0; cents >= 25; ++i)
    {
        cents = (cents - 25);
        ++ quarters ;
    }
    return quarters;
}

int calculate_dimes(int cents)
{
    int dimes = 0, i;

    for (i = 0; cents >= 10; ++i)
    {
        cents = (cents - 10);
        ++ dimes ;
    }
    return dimes;
}

int calculate_nickels(int cents)
{
    int nickels = 0, i;

    for (i = 0; cents >= 5; ++i)
    {
        cents = (cents - 5);
        ++ nickels ;
    }
    return nickels;
}

int calculate_pennies(int cents)
{
    int pennies = 0, i;

    for (i = 0; cents >= 1; ++i)
    {
        cents = (cents - 1);
        ++ pennies ;
    }
    return pennies;
}
